Zotero/Scenes/Detail/PDF/Views/PDFAnnotationsViewController.swift

```diff
@@ -181,96 +188,98 @@ final class PDFAnnotationsViewController: UIViewController {
-                snapshot.appendSections([0])
-                snapshot.appendItems(state.sortedKeys)
-                if let keys = state.updatedAnnotationKeys {
-                    snapshot.reloadItems(keys)
+            if state.changes.contains(.annotations) {
+                if state.changes.contains(.sidebarEditing) {
+                    tableView.setEditing(state.sidebarEditingEnabled, animated: false)
                 }
-                dataSource.apply(snapshot, animatingDifferences: isVisible, completion: completion)
+                updateUI(state: state, animatedDifferences: isVisible, completion: completion)
+                return
             }
-            return
-        }
 
-        if state.changes.contains(.interfaceStyle) {
-            updateQueue.async { [weak self] in
-                guard let self else { return }
-                var snapshot = dataSource.snapshot()
-                guard !snapshot.sectionIdentifiers.isEmpty else { return }
-                snapshot.reloadSections([0])
-                dataSource.apply(snapshot, animatingDifferences: false, completion: completion)
+            if state.changes.contains(.interfaceStyle) {
+                if dataSource.snapshot().numberOfItems > 0 {
+                    updateQueue.async { [weak self] in
+                        guard let self else { return }
+                        var snapshot = dataSource.snapshot()
+                        snapshot.reconfigureItems(snapshot.itemIdentifiers)
+                        dataSource.apply(snapshot, animatingDifferences: false, completion: completion)
+                    }
+                } else {
+                    completion()
+                }
+                return
             }
-            return
-        }
 
-        if state.changes.contains(.selection) || state.changes.contains(.activeComment) {
-            if let keys = state.updatedAnnotationKeys {
-                updateQueue.sync { [unowned self] in
+            if state.changes.contains(.selection) || state.changes.contains(.activeComment) {
+                updateQueue.async { [weak self] in
+                    guard let self else { return }
                     var snapshot = dataSource.snapshot()
-                    snapshot.reloadItems(keys)
-                    dataSource.apply(snapshot, animatingDifferences: false)
+                    let itemIdentifiers = snapshot.itemIdentifiers
+                    snapshot.reconfigureItems(itemIdentifiers)
+                    if let keys = state.updatedAnnotationKeys {
+                        snapshot.reloadItems(keys)
+                    }
+                    dataSource.apply(snapshot, animatingDifferences: false) { [weak self] in
+                        guard let self else { return }
+                        focusSelectedCell()
+                        if state.changes.contains(.sidebarEditing) {
+                            tableView.setEditing(state.sidebarEditingEnabled, animated: isVisible)
```

Comment:
I just realised, we apply `tableView.setEditing(` before the data source update on lines 205-207. Can this just be moved above?

Reply:
Check https://github.com/zotero/zotero-ios/pull/1040/commits/3af6442736fc7f6c1a61aa3cc91dc775ed033636